testing/skia_gold_client/lib/skia_gold_client.dart

```diff
@@ -55,9 +55,12 @@ class SkiaGoldClient {
   String get _keysPath => path.join(workDirectory.path, 'keys.json');
   String get _failuresPath => path.join(workDirectory.path, 'failures.json');
 
-  /// Indicates whether the `goldctl` tool has been initialized for the current
-  /// test context.
-  bool _isInitialized = false;
+  Future<void>? _initResult;
+  Future<void> _initOnce(Future<void> Function() callback) async {
```

Comment:
nit: the function body doesn't need to be marked as async, as you're not actually using `await` semantics or anything here, you're just directly returning something that is already a future.

Reply:
Done